src/Microsoft.AspNet.Mvc.TagHelpers/HiddenFormTagHelper.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Mvc.ViewFeatures;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting all form elements to generate hidden tag for checkboxes.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Advanced)]
+    [HtmlTargetElement("form")]
+    public class HiddenFormTagHelper : TagHelper
+    {
+        [HtmlAttributeNotBound]
+        [ViewContext]
+        public ViewContext ViewContext { get; set; }
```

Comment:
Actually, fine if order relative to `FormTagHelper` is not predetermined. `Order` should be `-1000` to keep execution away from user tag helpers.